subprojects/functional/src/main/java/org/gradle/internal/io/IoPredicate.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.io;
```

Comment:
How about using `gradle.internal.function` or `gradle.internal.function.io` as a package? `io` seems to be very generic.

Reply:
The whole package is about functional programming in a way (well, most of it is). We could move everything into `org.gradle.internal.functional` or something, though I wouldn't mix that in this PR.